bindings/python/scripts/generate.py

```diff
@@ -1,105 +1,417 @@
+            "FOR_STMT": handled_by_pybind,
+            "DECL_STMT": unsure,  # handled_by_pybind
+            "SWITCH_STMT": handled_by_pybind,
+            "CASE_STMT": handled_by_pybind,
+            "DEFAULT_STMT": handled_by_pybind,
+            "CXX_TRY_STMT": handled_by_pybind,
+            "CXX_CATCH_STMT": handled_by_pybind,
+            # TEMPLATE related
+            "CLASS_TEMPLATE": self.skip,  # self.handle_class_template
+            "TEMPLATE_NON_TYPE_PARAMETER": handled_elsewhere,  # in (handle_class_template)
+            "FUNCTION_TEMPLATE": self.skip,  # to be added later
+        }
 
+        self.handle_node(root)
 
-in_struct = False
-module_linelist = []
+    def skip(self):
+        """
+        Used to keep track of skipped elements, for debugging purposes.
 
+        Skipped elements can be:
+            - elements which are not handled in their own function, or
+            - elements which are not handled at all (skipped).
+        """
 
-def handle_final(filename, module_name):
-    linelist = []
-    linelist.append(f"#include <{filename}>")
-    linelist.append("#include <pybind11/pybind11.h>")
-    linelist.append("namespace py = pybind11;")
-    for i in range(len(module_linelist)):
-        if module_linelist[i].startswith("namespace"):
-            continue
-        else:
-            module_linelist[i] = "".join(
-                (f"PYBIND11_MODULE({module_name}, m)", "{", module_linelist[i])
+        self._skipped.append(
+            {
+                "line": self.item["line"],
+                "column": self.item["line"],
+                "kind": self.kind,
+                "name": self.name,
+            }
+        )
+
+    def end_scope(self):
+        """
+        Used for adding ending characters (braces, semicolons, etc.) when state's scope ends.
+        """
+
+        if self._state_stack[-1]["kind"] == "NAMESPACE":
+            self._linelist.append("}")
+        elif self._state_stack[-1]["kind"] == "STRUCT_DECL":
+            self._linelist.append(";")
+        elif self._state_stack[-1]["kind"] == "STRUCT_DECL":
+            self._linelist.append(";")
+        elif self._state_stack[-1]["kind"] == "CLASS_TEMPLATE":
+            self._linelist.append(";")
+
+    @staticmethod
+    def get_fields_from_anonymous(item):
+        """
+        Helper function to extract fields from anonymous types.
+
+        Parameters:
+            - item (dict): the anonymous type item from which to extract fields
+        
+        Returns:
+            - fields (list): A list of items of kind `CursorKind.FIELD_DECL`
+        """
+
+        fields = []
+        for sub_item in item["members"]:
+            # base condition
+            if sub_item["kind"] == "FIELD_DECL":
+                fields.append(sub_item)
+            # recurse
+            elif sub_item["kind"] in ("ANONYMOUS_UNION_DECL", "ANONYMOUS_STRUCT_DECL"):
+                fields += bind.get_fields_from_anonymous(sub_item)
```

Comment:
why change from self to bind?
```py
class A:
    def __init__(self):
        self.say_hi()
    @staticmethod
    def say_hi():
        print("I'm a static method, short and stout")

A()
A.say_hi()
```

_Originally posted by @kunaltyagi in https://github.com/divmadan/pcl/pull/8_

Reply:
It's being called from a class function:
https://github.com/divmadan/pcl/blob/d7f8f4b67d8dff6c4981698927bdd1bcdc1852ac/bindings/python/scripts/generate.py#L128-L148

Why create an object of class within the function?

Reply:
Stupid pylint!
It works without any `self` or classname :facepalm: 

Updated.